packages/queued-request-controller/tests/QueuedRequestMiddleware.test.ts

```diff
@@ -0,0 +1,425 @@
+    );
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  describe('confirmations', () => {
+    it('should resolve requests that require confirmations for infura networks', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware({
+        messenger,
+        useRequestQueue: () => true,
+      });
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        ...requestDefaults,
+        method: 'eth_sendTransaction',
+      };
+
+      await new Promise((resolve, reject) =>
+        middleware(req, {} as any, resolve, reject),
+      );
+
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      // infura networks do not use getNetworkClientById
+      expect(mocks.getNetworkClientById).not.toHaveBeenCalled();
+    });
+
+    it('should resolve requests that require confirmations for custom networks', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware({
+        messenger,
+        useRequestQueue: () => true,
+      });
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        ...requestDefaults,
+        networkClientId: 'custom-rpc.com',
+        method: 'eth_sendTransaction',
+      };
+
+      await new Promise((resolve, reject) =>
+        middleware(req, {} as any, resolve, reject),
+      );
+
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      // custom networks use getNetworkClientyId
+      expect(mocks.getNetworkClientById).toHaveBeenCalled();
+    });
+
+    it('switchEthereumChain calls get queued but we dont check the current network', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware({
+        messenger,
+        useRequestQueue: () => true,
+      });
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        ...requestDefaults,
+        method: 'wallet_switchEthereumChain',
+      };
+
+      await new Promise((resolve, reject) =>
+        middleware(req, {} as any, resolve, reject),
+      );
+
+      expect(mocks.addRequest).not.toHaveBeenCalled();
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      expect(mocks.getProviderConfig).not.toHaveBeenCalled();
+    });
+
+    describe('requiring switch', () => {
+      it('calls addRequest to switchEthChain if the current network is different than the globally selected network', async () => {
+        const messenger = buildMessenger();
+        const middleware = createQueuedRequestMiddleware({
+          messenger,
+          useRequestQueue: () => true,
+        });
+        const mockGetProviderConfig = jest.fn().mockReturnValue({
+          providerConfig: {
+            chainId: '0x5',
+          },
+        });
+        const mocks = buildMocks(messenger, {
+          getProviderConfig: mockGetProviderConfig,
+        });
+
+        const req = {
+          ...requestDefaults,
```

Comment:
[nit] adding a comment just to make it apparent that the chainId is different for the request and the currently selected network
```suggestion
          ...requestDefaults, // chainId = '0x1'
```